frontend/components/artists/artist_show.jsx

```diff
@@ -36,12 +38,15 @@ const ArtistShow = ({
 					backgroundImage: `url(${currentArtist.photoUrl})`,
 				}}
 			>
-				<div className="artist-show" id="artist-show">
+				<div className="artist-show" ref={artistShowRef}>
```

Comment:
This is to prevent the scrolling once the dropdown is open? Is there a way to design it so that the child isn't manipulating the parent / ancestor via ref, maybe with state?

Reply:
Interesting question. I specifically chose ref because I don't need the ArtistShow (ancestor) div to re-render since nothing changed in the view, while changing state would definitely re-render the component (and therefore all its chidlren). I just needed to toggle a CSS property on or off, which doesn't have to deal with React. Does this make sense?

Thanks for this comment because I copied this code over from my other dropdown, where it is working just fine, but just found a bug here where the ability to scroll doesn't return to ArtistShow.

Reply:
Fixed the bug. I moved the CSS-modifying code out of the dropdown to its parent because once the dropdown unmounts, its code isn't read again and the else to restore the scrolling functionality wasn't read.

Since ArtistPageMenuBar is the actual owner and parent of the dropdown, it has a useEffect that re-renders it according to dropdown state, so now the code modifying the CSS of the ancestor ArtistShow will always be re-read when the dropdown is toggled.

Reply:
Could you use your own hook? https://blog.logrocket.com/create-advanced-scroll-lock-react-hook/

Reply:
I'll definitely look into this as a way to create a custom Hook. Are you against the useRef because I've needed this functionality in multiple components so makes sense to extract?

I appreciate the link,

Reply:
I'm not as familiar with it so I'm trying to understand how you made the decision here

Reply:
`useRef` can be used to hold any value and have it persist throughout renders, but what I've read is that in terms of React, it can be used to refer to an HTML element instead of using vanilla JS (eg. `getElementById()`) since it's not advised to manipulate the DOM directly when using React.

Let me know if this helps? There are some useRef use cases following this point in the article: https://blog.logrocket.com/usestate-vs-useref/#:~:text=Consider%20situations%20where,or%20performance%20problems.

I chose to use `useRef` for a combination of the first two use cases mentioned in the article. I had no need to re-render the Artist Page because there is no data change (I am only tagging on a CSS property), and I needed a way to grab that HTML element specifically and act on it in a way that agrees with React.